moses/moses/main/table-problems.cc

```diff
@@ -106,62 +110,92 @@ void table_problem_base::common_setup(problem_params& pms)
         log_output_error_exit(ss.str());
     }
 
-    // Read input data files
-    _tables.clear();
-    _ctables.clear();
-    size_t num_rows = 0;
-    for (const string& idf : _tpp.input_data_files) {
-        logger().info("Read data file %s", idf.c_str());
-        Table table = loadTable(idf, _tpp.target_feature,
-                                _tpp.timestamp_feature,
-                                _tpp.ignore_features_str);
-        num_rows += table.size();
-
-        // Possibly subsample the table
-        if (pms.nsamples > 0)
-            subsampleTable(pms.nsamples/(float)table.size(), table);
-
-        // Compressed table
-        _ctables.push_back(table.compressed(_tpp.weighting_feature));
-
-        // The compressed table removes the weighting feature, too.
-        if (not _tpp.weighting_feature.empty())
-            table.itable.delete_column(_tpp.weighting_feature);
-        _tables.push_back(table);
-
-        // Possibly balance the ctable
-        if (pms.balance)
-            _ctables.back().balance();
-    }
-    logger().info("Number of rows in tables = %d", num_rows);
-
-    // XXX FIXME -- the multiple tables should be merged into one.
-    ctable = _ctables.front();
-    table = _tables.front();
-
-    // Get the labels contained in the data file.
-    ilabels.clear();
-    if (pms.output_with_labels)
-        ilabels = table.itable.get_labels();
-
-    arity = table.get_arity();
-
-    // Check that all input data files have the same arity
-    // XXX FIXME .. check that they all have the same signature.
-    if (_tables.size() > 1) {
-        for (size_t i = 1; i < _tables.size(); ++i) {
-            combo::arity_t test_arity = _tables[i].get_arity();
-            if (test_arity != arity) {
-                stringstream ss;
-                ss << "File " << _tpp.input_data_files[0] << " has arity " << arity
-                   << " while file " << _tpp.input_data_files[i] << " has arity " << test_arity;
-                log_output_error_exit(ss.str());
-            }
-        }
-    }
-    logger().info("Inferred arity = %d", arity);
+	// Read input data files
+    if(_tpp.use_atomese)
+		read_atomese_table(pms);
+    else
+        read_combo_table(pms);
```

Comment:
There's a mixture of spaces and tabs here. You want to respect the same indentation format within the same file. So you may either use whitespaces or reformat the entire table-problems.cc to use tabs, it's up to you (as long as such reformatting takes place in a different commit, within that PR, it's OK).